### ISSUE TITLE
Zerotier not working on cellular data Android 1.14.0 - Samsung S21+

### ISSUE THREAD

**Issue by JaJe92** [zerotier/ZeroTierOne#2334]
This problem also happened on previous version prior 1.14 but on older versions of software it worked and all of sudden it stopped.
I really need this and I ran out of options to fix this.

- What you expect to be happening.
  To work as intended in both Wifi connection and cellular data.

- What is actually happening?
  Whenever I connect to 4G/5G Network, Zerotier refuses to just connect to the network no matter what I do.
  
- Any steps to reproduce the error.
  I tried to clean cache of app, force close, clear everything and reconfigure the settings on phone, then reinstall it completely and remove the device from zerotier network and add it again, still nothing.

-Any relevant console output or screenshots.
I attach screenshots on my phone while I am on 5G that is offline and as soon I get into wifi, it will connect to the network and works.

![Screenshot_20240710_170531_ZeroTier One](https://github.com/zerotier/ZeroTierOne/assets/49617419/e49eb856-f6a5-4560-8e7b-edaeb2cd08bd)
![Screenshot_20240710_170534_ZeroTier One](https://github.com/zerotier/ZeroTierOne/assets/49617419/0b03533b-7916-420f-b51a-8b7045a80711)
![Screenshot_20240710_170655_ZeroTier One](https://github.com/zerotier/ZeroTierOne/assets/49617419/279042bc-e2b7-4e6a-a7f3-a6f7bb17b014)

Sometimes on 5G it appear the lock key that is connected to vpn but still offline like below.

![Screenshot_20240710_170806_ZeroTier One](https://github.com/zerotier/ZeroTierOne/assets/49617419/9a44c868-eaec-44f3-8165-c50b5f689be8)

-What operating system and ZeroTier version. Please try the latest ZeroTier release. 
Samsung S21+ 5G
Android 14, One UI version 6.1
Kernel version 5.4.242-28575149-xxxxx

**Comment by joseph-henry** [comment]
Thanks for the report. Who is your service provider?

**Comment by JaJe92** [comment]
Orange Romania

**Comment by Ponkos22** [comment]
I have same problem with Orange Poland. With t-mobile on second phone is OK

**Comment by JaJe92** [comment]
> I have same problem with Orange Poland. With t-mobile on second phone is OK

Thank you for confirming this, I am unable to test with another provider as everyone I know is using Orange only. I'm more convinced that it's the ISP fault at this point and they did changed something on their network without informing public that the network have been changed as I haven't found any information on their website.

**Comment by joseph-henry** [comment]
Unfortunately it does look like your ISP is blocking either our endpoints or certain encrypted UDP traffic. It's hard to tell. I think our only recourse in cases like this is to just file a complaint ticket with your ISP and trying to get clarification. I'm sorry.

If you're in a situation where the node *can* come online but it doesn't talk to other nodes, you can try:

In other discussions with clients reporting ZeroTier not working over cell radios we've realized that some devices along the path might have insanely small MTUs. So if you change your ZeroTier network's MTU to something much lower it might allow connectivity at the expense of traffic. You can play with the number to find one that works best for you.

In Central, you can create an API token and use it like so:

```
curl -X POST -H "Authorization: token xxxxxxxxxx" https://api.zerotier.com/api/v1/network/<nwid> -d '{ "config": { "mtu": 1280 } }'
```

Where `<nwid>` needs to be replaced with your network ID.

Some people have reported that setting the MTU to something like 1280 really helps. But this will have a  noticeable performance impact.

**Comment by Pinkbyte** [comment]
I am having the similar issue (not working on cellular data after upgrade to 1.14.0)

**Comment by JaJe92** [comment]
> I am having the similar issue (not working on cellular data after upgrade to 1.14.0, works on Wi-Fi just fine) Rolled back to 1.12.0 for now

I've done the same, unfortunately none of the older version of Zerotier apk don't work on my case.

**Comment by mstevanak** [comment]
I have a similar problem: On wifi it's OK, on celluar data I'm getting request time out, occasionaly I get OK response.

**Comment by bremenpl** [comment]
Exact same problem, also Orange, Poland, phone...

**Comment by SnowyStarSky** [comment]
Maybe it is related to your phone's MTU settings? I'm a bit of a networking noob so unsure if it's related but I was able to solve a connectivity issue of accessing services running on my server via my phone using mobile data by lowering the server's MTU setting for the ZT adapter to 1300 (max MTU for phone using mobile data was about 1350) from the default 2800

**Comment by tomV81** [comment]
I have the same problem with connection via GSM (Vodafone CZ).  Zerotier is posting online but I can't connect.  Ping doesn't work either.  Zerotier currently in version 1.14.0-2, but also tested version 1.10.  Everything is fine on Wi-Fi.

**Comment by boykov** [comment]
I have a similar problem with connection via GSM (Megafon). Ping doesn't work, OFFLINE status. Version 1.14.0-2. It works on Wi-Fi. A laptop (1.14.1) connected via smartphone Wi-Fi access point works fine too.

**Upd**. I somehow forced the app to work. Here are approximate steps:
- Just installed fresh app doesn't work via GSM, no ping, OFFLINE status
- Connect it via Wi-Fi and check the capacity for work: ONLINE status, connected to the network.
- Then using android "Applications" stop the app and clear the cache (I think it's an analog of "purging" in the case of linux https://discuss.zerotier.com/t/zerotier-cli-shows-offline-zerotier-gui-shows-connected/138/5 )
- Change Wi-Fi to GSM
- Start the app, now it's ONLINE via GSM and ping works.

I'm not sure that these steps work in any case, but I've already got working app after few fresh installs. And it continues to work now.

**Comment by htf1987** [comment]
> I have a similar problem with connection via GSM (Megafon). Ping doesn't work, OFFLINE status. Version 1.14.0-2. It works on Wi-Fi. A laptop (1.14.1) connected via smartphone Wi-Fi access point works fine too.
> 
> **Upd**. I somehow forced the app to work. Here are approximate steps:
> 
>     * Just installed fresh app doesn't work via GSM, no ping, OFFLINE status
> 
>     * Connect it via Wi-Fi and check the capacity for work: ONLINE status, connected to the network.
> 
>     * Then using android "Applications" stop the app and clear the cache (I think it's an analog of "purging" in the case of linux https://discuss.zerotier.com/t/zerotier-cli-shows-offline-zerotier-gui-shows-connected/138/5 )
> 
>     * Change Wi-Fi to GSM
> 
>     * Start the app, now it's ONLINE via GSM and ping works.
> 
> 
> I'm not sure that these steps work in any case, but I've already got working app after few fresh installs. And it continues to work now.

Thank you very much! Following these steps helped me.

**Comment by podyxa** [comment]
> I have a similar problem with connection via GSM (Megafon). Ping doesn't work, OFFLINE status. Version 1.14.0-2. It works on Wi-Fi. A laptop (1.14.1) connected via smartphone Wi-Fi access point works fine too.
> 
> **Upd**. I somehow forced the app to work. Here are approximate steps:
> 
> * Just installed fresh app doesn't work via GSM, no ping, OFFLINE status
> * Connect it via Wi-Fi and check the capacity for work: ONLINE status, connected to the network.
> * Then using android "Applications" stop the app and clear the cache (I think it's an analog of "purging" in the case of linux https://discuss.zerotier.com/t/zerotier-cli-shows-offline-zerotier-gui-shows-connected/138/5 )
> * Change Wi-Fi to GSM
> * Start the app, now it's ONLINE via GSM and ping works.
> 
> I'm not sure that these steps work in any case, but I've already got working app after few fresh installs. And it continues to work now.

Having the same issue on GSM (T2), Works great on wifi, trying these steps doesn't help, the phone is telling me that the vpn is on, and zerotier one on android says that the status is online, however the web interface doesn't ping the phone.

**Comment by boykov** [comment]
> Having the same issue on GSM (T2), Works great on wifi, trying these steps doesn't help, the phone is telling me that the vpn is on, and zerotier one on android says that the status is online, however the web interface doesn't ping the phone.

I agree what android zerotier works on some 4G networks just unstable. My steps above don't give a reliable solution, the connection can break in any moment or doesn't work at all. My further investigation related with linux zerotier through 4G. (I can't configure rootless android zerotier with local.conf, so I use linux zerotier through 4G instead.)
I can monitor published addresses/ports with 

sudo zerotier-cli info -j

and I can suggest addresses/ports for clients with /var/lib/zerotier-one/local.conf https://docs.zerotier.com/config/ (vritual ... try).

My conclusion: manual intervention can help to establish a connection through 4G in any circumstances, but zerotier can't do this automatically. I think the reason is 4G networks with restricted NAT: zerotier algorithm can't handle such cases.

Now I use yggdrasil instead of zerotier.
